### ISSUE TITLE
No errors if `$SHELL` is executed directly

### ISSUE THREAD

**Issue by hyperupcall** [koalaman/shellcheck#2724]
#### For bugs
- All versions
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck shell=bash

$SHELL --version

$any_var --version
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Expected to have error that `$SHELL` is not quoted. Although the value is usually `/usr/bin/bash`, it could very well be some long directory that includes whitespace.

No warnings show up for `$any_var` either -  I suppose this was done because people combine the command path and its flags in the same variable and having a lint would cause more trouble than it's worth? Although I disagree, I'm not asking to change other variables since it's behavior is not unreasonable.
